### PR TITLE
Add READ_EXTERNAL_STORAGE permission, to fix script.kodi.android.update

### PR DESCRIPTION
**Pull request by drinfernoo**
## Description
This should allow the app to be granted both READ_EXTERNAL_STORAGE *and* WRITE_EXTERNAL_STORAGE permissions, hopefully solving the issue of Kodi not being able to install other *.apk* files, like in the official Kodi Android Installer add-on.

## Motivation and Context
I don't believe there are any open issues regarding this, but as of Nov 2018, the script.kodi.android.update add-on stopped working (first mention of the crash [here](https://forum.kodi.tv/showthread.php?tid=322783&pid=2786299#pid2786299), I believe), and was simply ["impaired due to an Android API change"](https://github.com/xbmc/repo-scripts/commit/b3a61b7904dc2b3831e34a4d0995388184e2d7ef). This "API change" came in late 2018, when Google Play began enforcing `targetSdkVersion="26"` across new apps (more details on that [here](https://www.androidpolice.com/2017/12/19/play-store-require-new-updated-apps-target-recent-api-levels-distribute-native-apps-64-bit-support/)), and when Oreo (Android 8.0) updates started hitting devices, which made them need to request the proper permissions at runtime.

## Types of change
- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
I did a quick test however it seems it still does not allow the add-on to start the apk install

**Comment by drinfernoo** [comment]
Do we have any idea what is causing this incompatibility, then? The functionality worked just fine (and in other add-ons besides script.kodi.android.update) until the late Leia betas and/or early RCs.

File: tools/android/packaging/xbmc/AndroidManifest.xml.in [hunk 1]
```diff
@@ -9,6 +9,7 @@
         android:minSdkVersion="21"
         android:targetSdkVersion="26" />
 
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
```
Comment (Rechi) [inline tools/android/packaging/xbmc/AndroidManifest.xml.in]: This won't fix anything.

> Any app that declares the WRITE_EXTERNAL_STORAGE permission is implicitly granted this permission.

https://developer.android.com/reference/android/Manifest.permission.html#READ_EXTERNAL_STORAGE

File: tools/android/packaging/xbmc/src/Splash.java.in [hunk 2]
```diff
@@ -137,7 +137,7 @@ public class Splash extends Activity
           mSplash.mTextView.setText("Asking for permissions...");
           mSplash.mProgress.setVisibility(View.INVISIBLE);
 
-          requestPermissions(new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE},
+          requestPermissions(new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE, Manifest.permission.READ_EXTERNAL_STORAGE},
```
Comment (Rechi) [inline tools/android/packaging/xbmc/src/Splash.java.in]: same

**Comment by drinfernoo** [comment]
So it seems that as of Oreo, apps must be granted the permission to "Install from unknown sources" individually, rather than the old system-wide setting. How do apps go about doing this?

**Comment by Lunatixz** [comment]
The issue is with the fileprovder paths:
https://stackoverflow.com/questions/38200282/android-os-fileuriexposedexception-file-storage-emulated-0-test-txt-exposed

**Comment by drinfernoo** [comment]
@Lunatixz Any ideas on a potential fix?

**Comment by phunkyfish** [comment]
PR closed as it appears to be obsolete. 

If the author would like it kept open please just request so on the PR. If you feel there is still merit in merging this PR you can request the label be changed to `PR Cleanup: Backburner`.

**Pull request closed without merging**